relay-profiling/src/android.rs

```diff
@@ -126,97 +126,46 @@ impl AndroidProfile {
     }
 }
 
-pub fn expand_android_profile(payload: &[u8]) -> Result<Vec<Vec<u8>>, ProfileError> {
-    let profile = parse_android_profile(payload)?;
-    let mut items: Vec<Vec<u8>> = Vec::new();
-
-    if profile.transactions.is_empty() && profile.has_transaction_metadata() {
-        match serde_json::to_vec(&profile) {
-            Ok(payload) => items.push(payload),
-            Err(_) => {
-                return Err(ProfileError::CannotSerializePayload);
-            }
-        };
+fn parse_profile(payload: &[u8]) -> Result<AndroidProfile, ProfileError> {
+    let mut profile: AndroidProfile =
+        serde_json::from_slice(payload).map_err(ProfileError::InvalidJson)?;
 
-        return Ok(items);
+    if profile.transactions.is_empty() && !profile.has_transaction_metadata() {
```

Comment:
This can be joined with the next `if` to simplify control flow:

```rust
if let Some(transaction) = profile.transactions.drain(..).next() {
    // ...
} else if !profile.has_transaction_metadata() {
    return Err(ProfileError::NoTransactionAssociated);
}
```